File: Next element with greater frequency.cpp
//Problem: Next Element with Greater Frequency
//geeksforgeeks problem of the day -> 8th July 2025
//C++ CODE
class Solution {
  public:
    vector<int> nextFreqGreater(vector<int>& arr) {
        //CodeGenius
        unordered_map<int,int>freq;
        for(int i=0;i<arr.size();i++)
        freq[arr[i]]++;
        
        vector<int>ans(arr.size(),-1);
        stack<int>st;
        
        for(int i=0;i<arr.size();i++){
            while(!st.empty() && freq[arr[i]]>freq[arr[st.top()]]){
                ans[st.top()]=arr[i];
                st.pop();
            }
            st.push(i);
        }
        return ans;
    }
};

File: Next element with greater_frequency.java
//Problem: Next element with greater frequency
//geeksforgeeks problem of the day -> 18th January 2026
//JAVA CODE
class Solution {
    public ArrayList<Integer> nextFreqGreater(int[] arr) {
        int n = arr.length;
        Map<Integer, Integer> freq = new HashMap<>();

        for (int num : arr) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }

        int[] res = new int[n];
        Arrays.fill(res, -1);
        Stack<Integer> st = new Stack<>();

        for (int i = 0; i < n; i++) {
            while (!st.isEmpty() && freq.get(arr[i]) > freq.get(arr[st.peek()])) {
                res[st.pop()] = arr[i];
            }
            st.push(i);
        }
        ArrayList<Integer> result = new ArrayList<>();
        for (int x : res) {
            result.add(x);
        }
        return result;
    }
}